tracks/ruby/exercises/leap/readme.md

```diff
@@ -0,0 +1,30 @@
+_Leap_ is one of the first side exercises, unlocked by _TwoFer_. 
+
+## Reasonable Solutions
+
+```ruby
+class Leap
+  year % 4 == 0 && (year % 100 != 0 || year % 400 == 0)
+end
+```
+Variants: `module` instead of `class`; instantiate class.  
+
+
+## Common suggestions
+Copied from the Python notes by @yawpitch :
+- There are just two cases that return true:
+  - a year is a multiple of 4 *and not** 100
+  - a year is a multiple of 4, 100, and 400
+- _Order of operations_ matter:
+  - 75% of all years *cannot* be leap years because they are not mulitples of 4; test `year % 4 == 0` first
+  - 98.97% of all years that are multiples of 4 are not multiples of 100; test `year % 100 != 0` second
+  - 1.03% of all years that are multiples of 4 are also multiples of 100 and 400; test `year % 400 == 0` third
+- _Order of evaluation_ matters:
+  ```ruby
+  year % 4 == 0 && year % 100 != 0 || year % 400 == 0
+  ```
+  _looks_ right, but will force a year like 999 to be checked for being a multiple of 400 unnecessarily
+- Eliminate duplicate work; no year should ever have to be checked multiple times for the same condition
+
+## Talking points
+* Logical operators `&&` `||` are more idiomatic than `return ... if ...`
```

Comment:
I think so. When I write a `return` statement I usually go back to review the code and see if I can somehow remove it, extracting to another method or simplifying the logic.

Unless it is a guard clause at the beginning of a method or a situation where multiple exit points are worth it, like the "Bob" exercise.

Reply:
@MarcosX  Re:  your other comment:
> Not sure where exactly to put it, but on some solutions there will be a bunch of nested ifs and elses with a return statement.  

Do you mean that that's not covered by this Talking Point? Because I think we address the same thing?

I kept it short here, because most Leap submissions I see do combine the conditionals in one way or the other. Three if statements are rare, in my experience. What do you think? 
Or do you have a suggestion for a change?

Reply:
Yeah I agree with you that it's not that common in Ruby and talking about using logical operators will get to the same result.

Leap solutions on the Javascript track usually have an if for each year validation, so the code ends up with 3 nested if/else. But that might just not be the case for Ruby.

Thanks for taking the time to discuss this :)

Reply:
Thanks for clarifying! I'll keep  an eye on upcoming solutions, and check in a while if those nested solutions happen a lot in Ruby too.